File: src/plugin/combat/magic/lunar/teleport/NorthArdougneTeleportSpell.java
package plugin.combat.magic.lunar.teleport;

import com.rs.game.entity.WorldTile;
import com.rs.game.plugin.combat.spell.type.TeleportSpellPlugin;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 7/28/2017
 */
public class NorthArdougneTeleportSpell extends TeleportSpellPlugin {
	
	@Override
	public int levelRequired() {
		return 76;
	}
	
	@Override
	public int[] runesRequired() {
		return arguments(ASTRAL_RUNE, 2, LAW_RUNE, 1, WATER_RUNE, 5);
	}
	
	@Override
	public WorldTile destination() {
		return new WorldTile(2614, 3347, 0);
	}
	
	@Override
	public int spellId() {
		return 69;
	}
	
	@Override
	public double exp() {
		return 76;
	}
	
	@Override
	public MagicBook book() {
		return MagicBook.LUNAR;
	}
}

File: src/main/java/plugin/interaction/combat/magic/lunar/teleport/NorthArdougneTeleportSpell.kt
package plugin.interaction.combat.magic.lunar.teleport

import game.content.plugin.combat.spell.type.TeleportSpellPlugin
import game.global.WorldTile
import utility.constants.MagicConstants
import utility.constants.MagicConstants.MagicBook

/**
 * @author Tyluur <itstyluur></itstyluur>@icloud.com>
 * @since 7/28/2017
 */
class NorthArdougneTeleportSpell : TeleportSpellPlugin {
    override fun levelRequired(): Int {
        return 76
    }

    override fun runesRequired(): IntArray {
        return arguments(MagicConstants.ASTRAL_RUNE, 2, MagicConstants.LAW_RUNE, 1, MagicConstants.WATER_RUNE, 5)
    }

    override fun destination(): WorldTile {
        return WorldTile(2614, 3347, 0)
    }

    override fun spellId(): Int {
        return 69
    }

    override fun exp(): Double {
        return 76.0
    }

    override fun book(): MagicBook {
        return MagicBook.LUNAR
    }
}